### PR TITLE
Add stored binary fields to static backwards compatibility indices tests

### PR DESCRIPTION
**Pull request by mikemccand**
Follow on from #21440.

This just adds stored binary fields to our back compat indices and ensures we can still load these fields.

This change is just for master, so only regenerates 5.0.{0,1,2} back compat indices; when I backport to 5.x and 2.x I'll regenerate the older versions too.  I'll also add a compressed binary field to the 1.x indices, and I confirmed this would have caught #21440.

### CODE REVIEW

File: core/src/test/java/org/elasticsearch/bwcompat/OldIndexBackwardsCompatibilityIT.java [hunk 1]
```diff
@@ -473,6 +477,21 @@ void assertAliasWithBadName(String indexName, Version version) throws Exception
         assertFalse(client().admin().indices().prepareAliasesExist(aliasName).get().exists());
     }
 
+    /**
+     * Make sure we can load previously separately stored binary fields.
```
Comment (jpountz) [inline core/src/test/java/org/elasticsearch/bwcompat/OldIndexBackwardsCompatibilityIT.java]: previously separately?

**Review by jpountz — APPROVED** [review]
LGTM

**Comment by jasontedor** [comment]
@mikemccand I think that you'll need to regenerate the 5.1.1 indices too.

**Comment by mikemccand** [comment]
> @mikemccand I think that you'll need to regenerate the 5.1.1 indices too.

Ahh yes, thanks for the heads up @jasontedor ... I'll do that.

**Review by jasontedor — APPROVED** [review]
LGTM.

**Pull request merged**